### ISSUE TITLE
TileGroup never renders the legend attribute

### ISSUE THREAD

**Issue by zgrossbart** [carbon-design-system/carbon-components-react#2358]
Look at the TileGroup example here:  http://react.carbondesignsystem.com/?selectedKind=Tile&selectedStory=Selectable&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel

The example has a `legend` attribute with a value of `Selectable Tile Group`, but that attribute is never added to the `fieldset` element for the TileGroup.  This means there's no way to set a legend for the TileGroup and that results in the following accessibility issue:

![Screen Shot 2019-05-10 at 7 40 26 AM](https://user-images.githubusercontent.com/664044/57524811-f0dc4700-72f6-11e9-96ae-72de9d1658ad.png)

**Comment by dakahn** [comment]
👋🏽 Thanks so much for reporting! I'm unable to reproduce your error. Could you tell me what DAP ruleset you're testing with?

**Comment by zgrossbart** [comment]
I see this with the `IBM Accessibility February 2019` ruleset.
